### PR TITLE
Inline r_swap_* functions [fix #4895]

### LINKED ISSUE #4895: inline r_swap*
Acording to libr/include/r_util.h line 757
```
757 /*swap*/ //inline?
```

we should inline the r_swap_ things

### PR DESCRIPTION
**Pull request by Svenito**
Changed r_swap functions to be inline as per #4895

### CODE REVIEW

**Comment by ret2libc** [comment]
AFAIR the compiler is already able to choose better than us when it needs to inline something or not. But maybe someone else knows more about this :)

**Comment by Svenito** [comment]
According to GCC docs: "a non-static inline function is always compiled on its own in the usual fashion." So I guess in this case we won't get any change in how these functions are compiled. 

Simply close this PR then I guess?

**Comment by ret2libc** [comment]
I think so. IIRC the only way to make a function really inline is to put it in an header file and define it static + inline.

**Comment by Svenito** [comment]
Yeah, that's correct. Adding `-finline-functions` to the compiler flags will tell the compiler to inline _simple enough_ functions automatically however

**Comment by ret2libc** [comment]
Also high level of optimisations should automatically tell the compiler to inline simple enough functions when possible.

**Comment by alvarofe** [comment]
This shouldn't be necessary at all since i created to use it in patch_relocs but you can accomplish the same with the new API `r_read_bleX` so i should take a look and drop this file in favor of `r_read_bleX`

**Comment by radare** [comment]
This is the theory. But in practice this never happens unless you pass some very specific flags or the finctions is really small and defined in the same object. 

The reason for inlining this is to avoid making rhash depend on rutil just for two swap calls

**Comment by radare** [comment]
My reason was not size or performance. Just avoid unnecessary deps and simplify build

File: libr/util/swap.c [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
 
-R_API ut64 r_swap_ut64(ut64 val) {
+R_API inline ut64 r_swap_ut64(ut64 val) {
     val = ((val << 8) & 0xFF00FF00FF00FF00ULL ) | ((val >> 8) & 0x00FF00FF00FF00FFULL );
     val = ((val << 16) & 0xFFFF0000FFFF0000ULL ) | ((val >> 16) & 0x0000FFFF0000FFFFULL );
     return (val << 32) | (val >> 32);
 }
 
-R_API st64 r_swap_st64(st64 val) {
+R_API inline st64 r_swap_st64(st64 val) {
```
Comment (radare) [inline libr/util/swap.c]: R_API = public
inline = not public

this is not correct. inlined functions should be defined static and implemented in the .h

Comment (Svenito) [inline libr/util/swap.c]: Fix coming.

**Comment by ret2libc** [comment]
LGTM

File: shlr/squashfs/src/Makefile [hunk 2]
```diff
@@ -97,11 +97,11 @@ INCLUDEDIR = -I.
 INSTALL_DIR = /usr/local/bin
 CFLAGS+=-I/usr/local/include -g
 
-MKSQUASHFS_OBJS = mksquashfs.o read_fs.o sort.o swap.o pseudo.o compressor.o \
+MKSQUASHFS_OBJS = mksquashfs.o read_fs.o sort.o pseudo.o compressor.o \
 	action.o
 
 UNSQUASHFS_OBJS = unsquashfs.o unsquash-1.o unsquash-2.o unsquash-3.o \
```
Comment (alvarofe) [inline shlr/squashfs/src/Makefile]: this swap.o refers to its own swap.c implementation (shlr/squashfs/src/swap.c) so i would maintain swap.o here

File: libr/include/r_util.h [hunk 3]
```diff
@@ -754,13 +754,38 @@ R_API const ut8 *r_uleb128_decode (const ut8 *data, int *datalen, ut64 *v);
...
+static inline ut32 r_swap_ut32(ut32 val) {
+    val = ((val << 8) & 0xFF00FF00 ) | ((val >> 8) & 0xFF00FF );
+    return (val << 16) | (val >> 16);
+}
+
+static inline st32 r_swap_st32(st32 val) {
+    val = ((val << 8) & 0xFF00FF00) | ((val >> 8) & 0xFF00FF );
+    return (val << 16) | ((val >> 16) & 0xFFFF);
+}
+
```
Comment (radare) [inline libr/include/r_util.h]: unnecessary newline

**Comment by radare** [comment]
thanks

**Pull request merged**